mapper/src/main/java/com/ibm/mapper/model/algorithms/MLKEM.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SonarQube Cryptography Plugin
+ * Copyright (C) 2024 IBM
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ibm.mapper.model.algorithms;
+
+import com.ibm.mapper.model.Algorithm;
+import com.ibm.mapper.model.KeyEncapsulationMechanism;
+import com.ibm.mapper.utils.DetectionLocation;
+import javax.annotation.Nonnull;
+
+public class MLKEM extends Algorithm implements KeyEncapsulationMechanism {
+    // https://en.wikipedia.org/wiki/Kyber
+    // https://pq-crystals.org/kyber/
+
+    private static final String NAME =
+            "ML-KEM"; // Kyber, Module-Lattice-Based Key-Encapsulation Mechanism
+
+    public MLKEM(@Nonnull DetectionLocation detectionLocation) {
+        super(NAME, KeyEncapsulationMechanism.class, detectionLocation);
```

Comment:
Add OID and Nist security level here, especially important for quantum safe algorithms. 

> ML-KEM has three standardized parameter sets, each with its own OID:
>
>    ML-KEM-512:
>    OID: 2.16.840.1.101.3.4.4.1
>    ML-KEM-768:
>    OID: 2.16.840.1.101.3.4.4.2
>    ML-KEM-1024:
>    OID: 2.16.840.1.101.3.4.4.3

https://csrc.nist.gov/projects/computer-security-objects-register/algorithm-registration#KEM

Reply:
Will be done later, as I first need to add detection rules for [KyberParameters](https://downloads.bouncycastle.org/java/docs/bcprov-jdk18on-javadoc/org/bouncycastle/pqc/crypto/crystals/kyber/KyberParameters.html)